bamboo/bootstrap-tf-deployment.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+set -ex
+. ./bamboo/abort-if-not-pr-or-redeployment.sh
+. ./bamboo/abort-if-skip-integration-tests.sh
+. ./bamboo/abort-if-not-terraform.sh
+
+npm config set unsafe-perm true
+npm install
+. ./bamboo/set-bamboo-env-variables.sh
+
+if [[ $USE_NPM_PACKAGES == true ]]; then
+  echo "***Deploying stack with NPM packages"
+  (cd example && npm install)
+else
+  echo "***Deploying stack with built packages"
+  npm run bootstrap
+fi
+
+echo "Locking stack for deployment $DEPLOYMENT-tf"
+
+cd example
+set +e
+
+# Fetch terraform binary
+if ! curl -o terraform_${TF_VERSION}_linux_amd64.zip https://releases.hashicorp.com/terraform/${TF_VERSION}/terraform_${TF_VERSION}_linux_amd64.zip ; then
+  echo "ERROR: coudn't download terraform script" >&2
```

Comment:
Approach borrowed from CMR. They used a `.tf_version` file, though, whereas I used an env var. We already define many version dependencies in files, I'm happy to hear strong opinions on which approach is better.

Reply:
You probably *do* want to use a version file for something like this - conceivably we may want to have a build plan that builds against a previous version (e.g. backport/baseline branch) at the same time we're merging updates on a newer version of terraform.   Reading it from a .tf_version file makes it config-as code and *per branch*, doing it this way means you have to explicitly set it for each branch that's not on the same version as the master branch.

Reply:
5d3ed7d60 👍